src/adapters/adapter_test.ts

```diff
@@ -1,24 +1,90 @@
-import { Model } from "../model.ts";
-import { testDB } from "../testutils.ts";
-import { assertEquals } from "../../testdeps.ts";
-import { QueryBuilder } from "../querybuilder.ts";
+import { Model, FieldType } from '../model.ts';
+import { testDB } from '../testutils.ts';
+import { assertEquals } from '../../testdeps.ts';
+import { QueryBuilder } from '../querybuilder.ts';
+
+class User extends Model {
+  static tableName = 'users';
+  static fields = {
+    email: { type: FieldType.STRING },
+    age: { type: FieldType.NUMBER },
+    created_at: { type: FieldType.DATE },
+  };
+
+  email!: string;
+  age!: number;
+  created_at!: Date;
+}
+
+class Product extends Model {
+  static tableName = 'products';
+  static fields = {
+    name: { type: FieldType.STRING },
+  };
+
+  name!: string;
+}
 
 testDB(
-  "BaseAdapter: `addModel` should populate `adapter` property",
+  'BaseAdapter: `addModel` should populate `adapter` property',
   (client) => {
-    class User extends Model {
-      static tableName = "users";
-    }
     client.addModel(User);
 
     assertEquals(User.adapter, client);
-  },
+  }
+);
+
+testDB(
+  'BaseAdapter: `getAllModels` should return an array containing all classes of the registered Models ',
+  (client) => {
+    client.addModel(User);
+    client.addModel(Product);
+
+    const models = client.getAllModels();
+
+    assertEquals(models.length, 2);
+    assertEquals(models[0], User);
+    assertEquals(models[1], Product);
+  }
+);
+
+testDB(
+  'BaseAdapter: `truncateAllModels` should truncate all registered model tables',
+  async (client) => {
+    const date = new Date('5 June, 2020');
+
+    client.addModel(User);
+    client.addModel(Product);
+
+    await User.insert({
+      email: 'a@b.com',
+      age: 16,
+      created_at: date,
+    });
+
+    await User.insert({
+      email: 'b@c.com',
+      age: 16,
+      created_at: date,
+    });
+
+    await Product.insert({ name: 'notebook' });
+    await Product.insert({ name: 'pen' });
+
+    client.truncateAllModels();
```

Comment:
I found the reason why your test failed.

This method triggers a promise, and if you don't wait for a promise inside a test case, it will throw an error.

Here is the solution.

```ts
await client.truncateAllModels()
```

But, in order to do that, the method itself needs to return the promise, as I mentioned in the previous comment.

Other than that, we're ready to merge! 😁

Reply:
After I added this, it's now simply freezing the test

Reply:
```
cotton-tests | test [sqlite] BaseAdapter: 'truncateModels' should truncate all registered model tables ... ok (3ms)
```

Reply:
I see, it seems like the example that I give you doesn't work.

This one works:

```ts
public async truncateModels(): Promise<void> {
  for (const model of this.models) {
    await model.truncate();
  }
}
```

Reply:
Worked! Sorry for the trouble, I just started learning typescript a week ago. I'm originally a C programmer haha